servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1429,6 +1429,78 @@ Size2 TextureStorage::texture_size_with_proxy(RID p_proxy) {
 	return texture_2d_get_size(p_proxy);
 }
 
+void TextureStorage::texture_rd_initialize(RID p_texture, const RID &p_rd_texture) {
+	ERR_FAIL_COND(!RD::get_singleton()->texture_is_valid(p_rd_texture));
+
+	// TODO : investigate if we can support this, will need to be able to obtain the order and obtain the slice info
+	ERR_FAIL_COND_MSG(!RD::get_singleton()->texture_is_shared(p_rd_texture), "Please create the texture object using the original texture");
```

Comment:
So for now I've disabled this seeing we're going to create shared textures from this. May change this depending on our discussion around whether we're going to take ownership of the resource or not.

Reply:
I don't think we should support creating RS::textures from shared textures. The relationships will get every complex if we can:
Create a texture: A
create a view from texture A: B
Create a texture from B: C
create a view from C: D

Although maybe it will be possible to support in the future.

I think for now the error is the best approach

Reply:
Cool, and once you have the texture you can always use the RS methods for creating a shared texture, but honestly I can't really see a use case here. The RS::texture will be used primarily for display with shaders and shaders can already select the required layer and/or mipmap level. Generally slices are created to render to a specific layer and you would keep that on the RD side.